server/tests/trim_gap_handling.rs

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) 2024 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::net::SocketAddr;
+use std::time::Duration;
+
+use enumset::enum_set;
+use futures_util::StreamExt;
+use googletest::fail;
+use hyper_util::rt::TokioIo;
+use tempfile::TempDir;
+use tokio::io;
+use tokio::net::UnixStream;
+use tonic::codec::CompressionEncoding;
+use tonic::transport::{Channel, Endpoint, Uri};
+use tower::service_fn;
+use tracing::level_filters::LevelFilter;
+use tracing::{error, info};
+use url::Url;
+
+use restate_admin::cluster_controller::protobuf::cluster_ctrl_svc_client::ClusterCtrlSvcClient;
+use restate_admin::cluster_controller::protobuf::{
+    ClusterStateRequest, CreatePartitionSnapshotRequest, TrimLogRequest,
+};
+use restate_local_cluster_runner::{
+    cluster::Cluster,
+    node::{BinarySource, Node},
+};
+use restate_types::config::{LogFormat, MetadataStoreClient};
+use restate_types::logs::metadata::ProviderKind::Replicated;
+use restate_types::net::AdvertisedAddress;
+use restate_types::protobuf::cluster::node_state::State;
+use restate_types::retries::RetryPolicy;
+use restate_types::{config::Configuration, nodes_config::Role};
+
+mod common;
+
+#[tokio::test]
```

Comment:
I am using this rather than `test(restate_core::test)` because that macro just exits the process on panic, which prevents unwinding from running - and can lead to leaked spawned processes on test failure.